toolbox/src/DiscreteFilter.cpp

```diff
@@ -125,7 +132,10 @@ bool DiscreteFilter::initialize(BlockInformation* blockInfo, wbt::Error* error)
             }
             return 1;
         }
-        filter = new iCub::ctrl::Filter(*num, *den);
+        for (unsigned i = 0; i < inputSignalWidth; ++i) {
```

Comment:
The filter class support vector input/output, why are are you using a different filter for each component of the vector?